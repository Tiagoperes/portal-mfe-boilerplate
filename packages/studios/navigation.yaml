+ root ~ root.studios (/studios):
  like: string
  limit: number
  + studio (/{studioId}):
    + stacks (/stacks):
      tech: array (string[])
      + stack (/{stackId}):
        + starters (/starters):
          + starter (/{starterId}):
