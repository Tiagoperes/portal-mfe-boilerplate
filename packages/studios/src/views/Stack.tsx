import { Breadcrumb } from 'containers/Breadcrumb'
import { ViewPropsOf } from 'navigation'

export const Stack = ({ route, params }: ViewPropsOf<'root.studio.stacks.stack'>) => (
  <>
    <Breadcrumb />
    <p>Stack {params.stackId} of Studio {params.studioId}</p>
    <a href={route.starters.$link()}>Go to Starters</a>
    <a href={route.$parent.$link()}>Go back to Stacks</a>
  </>
)
