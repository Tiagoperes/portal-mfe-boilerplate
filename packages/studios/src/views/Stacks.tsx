import { Breadcrumb } from 'containers/Breadcrumb'
import { ViewPropsOf } from 'navigation'

export const Stacks = ({ route, params }: ViewPropsOf<'root.studio.stacks'>) => (
  <>
    <Breadcrumb />
    <p>Stacks of Studio {params.studioId}. Tech: {params.tech?.join(', ')}</p>
    <a href={route.stack.$link({ stackId: '1' })}>Go Stack 1</a>
    <a href={route.stack.$link({ stackId: '2' })}>Go Stack 2</a>
    <a href={route.$parent.$link()}>Go back to Studio</a>
  </>
)
