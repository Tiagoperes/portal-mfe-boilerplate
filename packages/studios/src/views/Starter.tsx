import { Breadcrumb } from 'containers/Breadcrumb'
import { ViewPropsOf } from 'navigation'

export const Starter = ({ route, params }: ViewPropsOf<'root.studio.stacks.stack.starters.starter'>) => (
  <>
    <Breadcrumb />
    <p>Starter {params.starterId} of Stack {params.stackId} of Studio {params.studioId}</p>
    <a href={route.$parent.$link()}>Go back to Starters</a>
  </>
)
