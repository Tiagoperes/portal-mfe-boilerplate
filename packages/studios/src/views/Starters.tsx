import { Breadcrumb } from 'containers/Breadcrumb'
import { ViewPropsOf } from 'navigation'

export const Starters = ({ route, params }: ViewPropsOf<'root.studio.stacks.stack.starters'>) => (
  <>
    <Breadcrumb />
    <p>Starters of Stack {params.stackId} of Studio {params.studioId}</p>
    <a href={route.starter.$link({ starterId: '1' })}>Go Starter 1</a>
    <a href={route.starter.$link({ starterId: '2' })}>Go Starter 2</a>
    <a href={route.$parent.$link()}>Go back to Stack</a>
  </>
)
