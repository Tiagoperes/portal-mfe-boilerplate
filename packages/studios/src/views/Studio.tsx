import { Breadcrumb } from 'containers/Breadcrumb'
import { ViewPropsOf } from 'navigation'

export const Studio = ({ route, params }: ViewPropsOf<'root.studio'>) => (
  <>
    <Breadcrumb />
    <p>Studio {params.studioId}</p>
    <a href={route.stacks.$link()}>Go to Stacks</a>
    <a href={route.$parent.$link()}>Go back to Studios</a>
  </>
)
