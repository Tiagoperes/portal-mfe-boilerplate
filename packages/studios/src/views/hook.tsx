import { useNavigationContext } from 'navigation'
import { ReactElement, useState } from 'react'
import { Stack } from './Stack'
import { Stacks as StudioStacks } from './Stacks'
import { Starter } from './Starter'
import { Starters } from './Starters'
import { Studio } from './Studio'
import { Studios } from './Studios'

export function useContent() {
  const [content, setContent] = useState<ReactElement>()
  
  useNavigationContext((context) => {
    context
      .when('root', props => setContent(<Studios {...props} />))
      .when('root.studio', props => setContent(<Studio {...props} />))
      .when('root.studio.stacks', props => setContent(<StudioStacks {...props} />))
      .when('root.studio.stacks.stack', props => setContent(<Stack {...props} />))
      .when('root.studio.stacks.stack.starters', props => setContent(<Starters {...props} />))
      .when('root.studio.stacks.stack.starters.starter', props => setContent(<Starter {...props} />))
      .whenNotFound(() => setContent(<h1>404: Not Found</h1>))
  })
    
  return content
}

// Default export is required by Module Federation
// eslint-disable-next-line import/no-default-export
export default useContent
