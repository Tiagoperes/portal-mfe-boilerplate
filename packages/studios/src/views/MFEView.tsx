/* This file is exported as a Module Federation Module */

import { useContent } from './hook'

export const MFEView = () => {
  const content = useContent()
  return content
}
