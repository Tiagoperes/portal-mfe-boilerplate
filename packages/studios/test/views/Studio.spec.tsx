import { CSSToCitricAdapter } from '@stack-spot/portal-theme'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render } from '@testing-library/react'
import { root } from 'navigation'
import { beforeAll, describe, expect, it } from 'vitest'
import { Studio } from 'src/views/Studio'

describe('Tests of Studio page', () => {
  const studioId = '1'

  beforeAll(() => {
    root.studio.$go({ studioId })
  })

  it('Should match the snapshot', () => {
    const queryClient = new QueryClient()
    const { container } = render(
      <CSSToCitricAdapter>
        <QueryClientProvider client={queryClient}>
          <Studio route={root.studio} params={{ studioId }} />
        </QueryClientProvider>
      </CSSToCitricAdapter>,
    )
    expect(container).toMatchSnapshot()
  })
})
