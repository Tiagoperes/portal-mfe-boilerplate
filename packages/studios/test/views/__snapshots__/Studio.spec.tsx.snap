// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests of Studio page > Should match the snapshot 1`] = `
.c1 {
  animation: cShZpT 1s linear infinite alternate;
  width: 100%;
  height: 21px;
}

.c0 {
  margin-bottom: 24px;
}

.c0 ul {
  display: flex;
  flex-direction: row;
}

.c0 li {
  display: flex;
  flex-direction: row;
}

.c0 li:before {
  content: '/';
  font: tag:p;
  weight: regular;
  size: 0.875rem;
  family: system2;
  line-height: 1.5;
  margin: 0 10px;
  color: var(--light-700);
}

.c0 li:first-child:before {
  content: none;
}

.c0 a {
  color: var(--light-700);
}

.c0 a:hover {
  text-decoration: underline;
  color: var(--light-contrastText);
}

<div>
  <nav
    class="c0"
  >
    <div
      class="c1"
      height="21px"
    />
  </nav>
  <p>
    Studio 
    1
  </p>
  <a
    href="/#/studios/1/stacks"
  >
    Go to Stacks
  </a>
  <a
    href="/#/studios"
  >
    Go back to Studios
  </a>
</div>
`;
